cmd/registry/cmd/annotate_test.go

```diff
@@ -0,0 +1,178 @@
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"testing"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestAnnotate(t *testing.T) {
+	var err error
+
+	const (
+		projectID   = "annotate-test"
+		projectName = "projects/" + projectID
+		apiID       = "sample"
+		apiName     = projectName + "/apis/" + apiID
+		versionID   = "1.0.0"
+		versionName = apiName + "/versions/" + versionID
+		specID      = "openapi.json"
+		specName    = versionName + "/specs/" + specID
+	)
+
+	// Create a registry client.
+	ctx := context.Background()
+	registryClient, err := connection.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("error creating client: %+v", err)
+	}
+	defer registryClient.Close()
+	// Clear the test project.
+	err = registryClient.DeleteProject(ctx, &rpc.DeleteProjectRequest{
+		Name: projectName,
+	})
+	if err != nil && status.Code(err) != codes.NotFound {
+		t.Fatalf("error deleting test project: %+v", err)
+	}
+	// Create the test project.
+	_, err = registryClient.CreateProject(ctx, &rpc.CreateProjectRequest{
+		ProjectId: projectID,
+		Project: &rpc.Project{
+			DisplayName: "Test",
+			Description: "A test catalog",
+		},
+	})
+	if err != nil {
+		t.Fatalf("error creating project %s", err)
+	}
+	// Create a sample api.
+	_, err = registryClient.CreateApi(ctx, &rpc.CreateApiRequest{
+		Parent: projectName,
+		ApiId:  apiID,
+		Api:    &rpc.Api{},
+	})
+	if err != nil {
+		t.Fatalf("error creating api %s", err)
+	}
+	// Create a sample version.
+	_, err = registryClient.CreateApiVersion(ctx, &rpc.CreateApiVersionRequest{
+		Parent:       apiName,
+		ApiVersionId: versionID,
+		ApiVersion:   &rpc.ApiVersion{},
+	})
+	if err != nil {
+		t.Fatalf("error creating version %s", err)
+	}
+	// Create a sample spec.
+	_, err = registryClient.CreateApiSpec(ctx, &rpc.CreateApiSpecRequest{
+		Parent:    versionName,
+		ApiSpecId: specID,
+		ApiSpec: &rpc.ApiSpec{
+			MimeType: "application/x.openapi;version=3.0.0",
+			Contents: []byte(`{"openapi": "3.0.0", "info": {"title": "test", "version": "v1"}, "paths": {}}`),
+		},
+	})
+	if err != nil {
+		t.Fatalf("error creating spec %s", err)
+	}
+
+	testCases := []struct {
+		comment  string
+		args     []string
+		expected map[string]string
+	}{
+		{comment: "add some annotations",
+			args:     []string{"a=1", "b=2"},
+			expected: map[string]string{"a": "1", "b": "2"}},
+		{comment: "remove one annotation and overwrite the other",
+			args:     []string{"a=3", "b-", "--overwrite"},
+			expected: map[string]string{"a": "3"}},
+		{comment: "changing an annotation without --overwrite should be ignored",
+			args:     []string{"a=4"},
+			expected: map[string]string{"a": "3"}},
+	}
+	// test annotations for APIs.
+	for _, tc := range testCases {
+		cmd := annotateCmd()
+		cmd.SetArgs(append([]string{apiName}, tc.args...))
```

Comment:
@seaneganx Notice this got simpler when I found that we don't need a rootCmd to invoke the subcommand. (This assumes that we're not using any inherited options, which might be a problem for some current or future commands, but we could address that by constructing the handler with `rootCmd()` instead.